crates/core/src/network/connection_manager.rs

```diff
@@ -93,14 +93,19 @@ impl Default for ConnectionManagerInner {
 pub struct ConnectionManager {
     inner: Arc<Mutex<ConnectionManagerInner>>,
     metadata: Metadata,
+    handshake_timeout: Duration,
 }
 
 impl ConnectionManager {
     /// Creates the connection manager.
-    pub(super) fn new(metadata: Metadata) -> Self {
+    pub(super) fn new(metadata: Metadata, handshake_timeout: Duration) -> Self {
```

Comment:
It might be better to future-proof this as pass in NetworkingOptions since this is a long-living object anyway.

Reply:
Yeah, i was thinking of that i just didn't want to make clones of it everywhere. And was honestly considering using an Arc for it (maybe with a type alias to reduce clutter) what do you think ?

Reply:
ConnectionManager is a long living value that lives as long as the node, cloning is absolutely fine here.